### PR TITLE
Speed up CRC map generation

### PR DESCRIPTION
**Pull request by dcvz**

### CODE REVIEW

File: extern/ZAPDUtils/Utils/StringHelper.cpp [hunk 1]
```diff
@@ -51,6 +82,26 @@ std::string StringHelper::Replace(std::string str, const std::string& from,
 	return str;
 }
 
+std::string StringHelper::Replace(std::string_view str, const std::string& from,
```
Comment (m4xw) [inline extern/ZAPDUtils/Utils/StringHelper.cpp]: Silly func of mine, we can rid it

File: src/resource/Archive.cpp [hunk 2]
```diff
@@ -304,17 +304,18 @@ bool Archive::LoadPatchMPQs() {
...
+    std::vector<std::string_view> lines = StringHelper::Split(std::string_view(listFile->Buffer.get(), listFile->BufferSize), "\n");
+
+    for (size_t i = 0; i < lines.size(); i++)
+    {
+        // Use std::string_view to avoid unnecessary string copies
+        std::string_view line = lines[i].substr(0, lines[i].length() - 1); // Trim \r
+        std::string line_str = std::string(line);
+
+        // Not NULL terminated str
+        uint64_t hash2 = ~crc64(line.data(), line.length());
```
Comment (m4xw) [inline src/resource/Archive.cpp]: At least from testing this didnt regress, only the forward slash hash seems required

File: src/resource/Archive.cpp [hunk 3]
```diff
@@ -304,17 +304,18 @@ bool Archive::LoadPatchMPQs() {
...
-        uint64_t hash2 = CRC64(StringHelper::Replace(line, "\\", "/").c_str());
-        mHashes[hash] = line;
-        mHashes[hash2] = line2;
+    // Use std::string_view to avoid unnecessary string copies
+    std::vector<std::string_view> lines = StringHelper::Split(std::string_view(listFile->Buffer.get(), listFile->BufferSize), "\n");
+
+    for (size_t i = 0; i < lines.size(); i++)
+    {
+        // Use std::string_view to avoid unnecessary string copies
+        std::string_view line = lines[i].substr(0, lines[i].length() - 1); // Trim \r
```
Comment (m4xw) [inline src/resource/Archive.cpp:313]: Tested on Win64 and MacOS, question is do we ever encounter non-crlf, ideally just handle it compile time as required

**Review by m4xw — APPROVED** [review]

**Review by m4xw — APPROVED** [review]

**Comment by dcvz** [comment]
Force pushed to point this at the `stable` branch instead

**Review by Kenix3 — APPROVED** [review]
Can you make a duplicate PR to main branch?

**Comment by Kenix3** [comment]
Actually I can cherry pick it locally.

**Pull request merged**